TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilder.cs

```diff
@@ -462,19 +469,15 @@ public virtual void StartDataConditionEntry(DataConditionEntry entry)
             if (_CurrentTypeDefinition != null)
                 node.ParentTypeDefinition = _CurrentTypeDefinition;
             Enter(node);
-            if (!node.IsPartOfATypeDef) node.SymbolTable.AddVariable(node);
+                node.SymbolTable.AddVariable(node);
         }
 
         public virtual void StartTypeDefinitionEntry(DataTypeDescriptionEntry typedef)
         {
             SetCurrentNodeToTopLevelItem(typedef.LevelNumber);
             var node = new TypeDefinition(typedef);
             Enter(node);
-            SymbolTable table;
-            if (node.CodeElement().IsGlobal) // TCTYPE_GLOBAL_TYPEDEF
```

Comment:
Comment `TCTYPE_GLOBAL_TYPEDEF` refers to a rule into our wiki, please add it back